aiChatterBox/aiChatterBox/MainWindow.xaml.cs

```diff
@@ -0,0 +1,284 @@
+﻿//Sauraav Jayrajh
+//sauraavjayrajh@gmail.com
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Input;
+using System.Windows.Media;
+
+namespace aiChatterBox
+{
+    public partial class MainWindow : Window
+    {
+        //Client to Handle API Interactions
+        private static readonly HttpClient client = new HttpClient();
+        //Adress of Ollama Running On Machine, Change if Needed
+        public static string localhost = "http://localhost:11434";
+        //Store Chats
+        private List<List<string>> chats = new List<List<string>>();
+        //Store current chat index
+        private int currentChatIndex = -1;
+        //To Indicate Start/Stop of Loading Animation
+        private CancellationTokenSource currentCancellationTokenSource = null;
+
+        //Colours for Chat Messages
+        public Brush userMessageColour = Brushes.Teal;
+        public Brush llamaMessageColour = Brushes.DarkBlue;
+        public Brush errorMessageColour = Brushes.Crimson;
+
+        //File to Store Chats
+        private string chatsFilePath = "pastChats.json";
+
+        public MainWindow()
+        {
+            InitializeComponent();
+            LoadChatsFromFile();
+        }
+
+        //Reuseable Method to Handle Submission of Prompt
+        public async void submitPrompt()
+        {
+
+            //Get and Validate Input
+            string inputPrompt = textBox_PromptInput.Text;
+            if (string.IsNullOrWhiteSpace(inputPrompt))
+            {
+                MessageBox.Show("Please enter a prompt");
+                return;
+            }
+
+            //Ensure current chat exists
+            if (currentChatIndex == -1)
+            {
+                currentChatIndex = chats.Count;
+                chats.Add(new List<string>());
+                listView_PastChats.Items.Add($"             Chat 1");
+            }
+
+            //Add User Input to Chat
+            chats[currentChatIndex].Add($"ME: {inputPrompt}");
```

Comment:
I understand why you doing this:
`chats[currentChatIndex]`

However would it not be easier to manage a private variable that holds that object and type so that it is easier to read and support, functionally the same.

```
private int currentChatIndex = -1;
private ControlType _chat;
private List<List<string>> chats = new List<List<string>>();


private void listView_PastChats_SelectionChanged(object sender, SelectionChangedEventArgs e)
{
}
```